modules/host/network.go

```diff
@@ -5,21 +5,57 @@ import (
 	"sync/atomic"
 	"time"
 
+	"github.com/NebulousLabs/Sia/build"
 	"github.com/NebulousLabs/Sia/encoding"
 	"github.com/NebulousLabs/Sia/modules"
 	"github.com/NebulousLabs/Sia/types"
 )
 
+// threadedUpdateHostname periodically runs 'managedLearnHostname', which
+// checks if the host's hostname has changed, and makes an updated host
+// announcement if so.
+//
+// TODO: This thread doesn't actually have clean shutdown, because the sleep is
+// outside of of the resource lock.
```

Comment:
Yeah, was thinking that. But then I need to add another field to the host, and I'm not sure how it's going to interact with the other functions/routines that don't seem to be closing cleanly right now.

I'm going to come back to this after all the features are written, then I can compose a closing strategy that takes into account everything that's happening in the host.